Entities/Activity.cpp

```diff
@@ -0,0 +1,1136 @@
+								m_ControlledActor[player]->AddAISceneWaypoint(actor->GetLastAIWaypoint());
+							}
+						}
+						actor->ClearAIWaypoints();
+						actor->SetAIMode(Actor::AIMODE_SQUAD);
+						actor->AddAIMOWaypoint(m_ControlledActor[player]);
+						// Make sure actor has m_ControlledActor registered as an AIMOWaypoint
+						actor->UpdateMovePath();
+					}
+					actor = g_MovableMan.GetNextTeamActor(team, actor);
+				} while (actor && actor != m_ControlledActor[player]);
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool Activity::SwitchToActor(Actor *actor, int player, int team) {
+		if (team < Activity::TEAM_1 || team >= Activity::MAXTEAMCOUNT || player < Activity::PLAYER_1 || player >= Activity::MAXPLAYERCOUNT) {
+			return false;
+		}
+		if (!actor || !g_MovableMan.IsActor(actor)) {
+			return false;
+		}
+		// Computer players don't focus on any Actor
+		if (!m_IsHuman[player]) {
+			return false;
+		}
+
+		// If the desired actor is not the brain and is controlled by someone else, 
+		// OR if it's actually a brain of another player, we can't switch to it
+		if (actor != m_Brain[player] && actor->IsPlayerControlled() || IsOtherPlayerBrain(m_ControlledActor[player], player)) {
+			g_GUISound.UserErrorSound()->Play();
+			return false;
+		}
+
+		const Actor *prevActor = 0;
+		// Disable human player control of the actor we're switching from
+		if (m_ControlledActor[player] && g_MovableMan.IsActor(m_ControlledActor[player])) {
+			// Set AI controller active to the guy we're switching from
+			m_ControlledActor[player]->SetControllerMode(Controller::CIM_AI);
+			m_ControlledActor[player]->GetController()->SetDisabled(false);
+			prevActor = m_ControlledActor[player];
+		}
+		// Switch
+		m_ControlledActor[player] = actor;
+
+		// Now set the controller and team of the actor we switched to.
+		if (m_ControlledActor[player]) {
+			m_ControlledActor[player]->SetTeam(team);
+			m_ControlledActor[player]->SetControllerMode(Controller::CIM_PLAYER, player);
+			m_ControlledActor[player]->GetController()->SetDisabled(false);
+
+			// Play actor switching sound effects
+			(m_ControlledActor[player] == m_Brain[player]) ? g_GUISound.BrainSwitchSound()->Play() : g_GUISound.ActorSwitchSound()->Play();
+			// Only play air swoosh if actors are out of sight of each other
+			if (prevActor && Vector(prevActor->GetPos() - m_ControlledActor[player]->GetPos()).GetMagnitude() > g_FrameMan.GetResX() / 2) { g_GUISound.CameraTravelSound()->Play(); }			
+
+			ReassignSquadLeader(player, team);
+		}
+		// Follow the new guy normally
+		m_ViewState[player] = NORMAL;
+
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void Activity::SwitchToNextActor(int player, int team, Actor *skip) {
```

Comment:
This method and SwitchToPrevActor are functionally identical, here's what needs to change:
1. They should both call SwitchToActor with the actor to switch to being g_MovableMan.GetNextTeamActor or GetPrevTeamActor
2. They still need to account for the actor to skip, so you need to check if the actor you're getting from movableman is the one to skip
3. They need to account for if SwitchToActor returns false and, if so, try the next/previous actor after that (i.e. stick it in a while loop)
4. They need to account for if we've looked through every actor without being able to swap to one, i.e. make sure to replicate the stuff under `if (m_ControlledActor[player] == start) {`

Also, I think SwitchToPrevActor should be above SwitchToNextActor, cause it just makes more sense imo.

Reply:
I'm gonna let you deal with this and the one above because I don't even.